tests/unit/src/test/scala/tests/SelectionRangeLspSuite.scala

```diff
@@ -0,0 +1,150 @@
+package tests
+import scala.collection.JavaConverters._
+
+/**
+ *  Suite for testing selection ranges. Note that this takes some liberty
+ *  since the spec isn't the clearest on the way this is done and different
+ *  LSP clients seems to impliment this differently. Mainly, we mimic what VS
+ *  Code does and what other servers like jdtls do and we send in a single
+ *  position. Once that positions is recieved we check the Selection range and
+ *  all the parents in that range to ensure it contains the positions we'd
+ *  expect it to contain. In the case of VS Code, they never make a second
+ *  request, and instead they just rely on the tree to continue selecting. So
+ *  we mimic that here in the tests.
+ */
+class SelectionRangeLspSuite extends BaseLspSuite("selectionRange") {
```

Comment:
We could add some test in `cross` since this is implemented using the compiler? We could reuse them later for Scala 3 and we can create a lot of smaller cases.

For example:
```scala
object Main extends App {
  val name: Option[String] = Option("Tom")
  name match {
    case Some(name) => println("Hello! " + name)
    case None =>
  }
}
```
This one seem particulary weird, since we get range on `case Some(name) => println("Hello! " + name)` and then with whitespaces before.

Is that maybe a tree node that we can skip?

Reply:
Or maybe the whitespaces selection is caused by VS Code? Anyway, I would for sure add cross tests with some edge cases as mentioned below.

Reply:
I _believe_ the whitespace all on that single line is VS Code either adding another range or something. I witnessed a lot of odd stuff when testing with VS Code that I can't reproduce when using coc-metals.

 
![2021-06-10 17 11 18](https://user-images.githubusercontent.com/13974112/121550510-16286b80-ca0f-11eb-9bd5-95474e5d8f8a.gif)

Either way, I'll for sure add some cross tests.

Reply:
I was afraid it was because of VS Code. They are probably trying to be smart and add another selection from the start of the line.